src/infill.cpp

```diff
@@ -234,6 +238,390 @@ void Infill::multiplyInfill(Polygons& result_polygons, Polygons& result_lines)
     }
 }
 
+void Infill::generateGyroidInfill(Polygons& result_lines)
+{
+    // generate infill based on the gyroid equation: sin_x * cos_y + sin_y * cos_z + sin_z * cos_x = 0
+    // kudos to the author of the Slic3r implementation equation code, the equation code here is based on that
+
+    if (zig_zaggify)
+    {
+        outline_offset -= infill_line_width / 2; // the infill line zig zag connections must lie next to the border, not on it
+    }
+
+    const Polygons outline = in_outline.offset(outline_offset);
+    const AABB aabb(outline);
+
+    int pitch = line_distance * 2.41; // this produces similar density to the "line" infill pattern
+    int num_steps = 4;
```

Comment:
The number 4. Why 4 and not 2, I am curious

Reply:
The line is based on a sine wave and so I figured that at the lowest resolution you still need 4 points to model a sine wave (0,+peak, 0, -peak).